prow/crier/reporters/slack/reporter.go

```diff
@@ -98,32 +107,38 @@ func (sr *slackReporter) GetName() string {
 
 func (sr *slackReporter) ShouldReport(pj *v1.ProwJob) bool {
 	logger := sr.logger.WithFields(pjutil.ProwJobFields(pj))
-	// if a user specifically put a channel on their job, they want
-	// it to be reported regardless of what other settings exist
-	if _, set := jobChannel(pj); set {
-		logger.Debugf("reporting as channel is explicitly set")
-		return true
+	jobCfg := jobConfig(pj)
+	prowCfg := sr.getConfig(pj)
+	// If a user specifically put a channel on their job, they want
+	// it to be reported regardless of the job types setting.
+	jobShouldReport := false
+	if jobCfg != nil && jobCfg.Channel != "" {
```

Comment:
Nit: Wouldn't all of this be easier expressed as `if jobCfg != nil && jobCfg.Channel != "" && (len(jobCfg.StatesToReport == 0 || stateSliceHasState(jobCfg.StatesToReport, job.Status.State) { return true }`?

A bit of a nitpicky comment but I had to read the tests to understand what you want to do

Reply:
It actually should be
```
if jobCfg != nil && jobCfg.Channel != "" && ((len(jobCfg.StatesToReport) == 0 && stateSliceHasState(prowCfg.StatesToReport, job.Status.State)) || stateSliceHasState(jobCfg.StatesToReport, job.Status.State)) { return true }
```
, which looks too long and not that expressive to me. I have updated the comments and hopefully it can explain the logic more clearly.

Reply:
You mean `len(jobCfg.StatesToReport) == 0 || stateSliceHasState(prowCfg.StatesToReport, job.Status.State)`?

The `|| stateSliceHasState(jobCfg.StatesToReport, job.Status.State)` part means its breaking though? Because today we report every job that has a channel regardless of state. I guess changing that makes sense, just wondering if it will surprise ppl

Reply:
Could you maybe add something to prow/ANNOUNCEMENTS.md in the breaking change section? I agree that changing this makes sense but I have no idea if there are maybe ppl who expect todays behavior.
cc @xrstf

Reply:
I have added this as a breaking change to `prow/ANNOUNCEMENTS.md`.

To provide more context, our use case is we only want to report the `failure` and `error` states for a few specific Prow jobs, without this change we won't be able to do so.

Reply:
Yeah, makes perfect sense. Thanks.